File: java/src/main/java/problems1_100/JumpGame.java
package problems1_100;

enum JumpGame {
    DEFAULT {
        @Override
        boolean canJump(int[] nums) {
            int goal = nums.length - 1;
            for (int i = nums.length - 1; i >= 0; i--) {
                if (i + nums[i] >= goal) {
                    goal = i;
                }
            }
            return goal == 0;
        }
    };

    abstract boolean canJump(int[] nums);
}

File: kotlin/src/main/kotlin/problems1_100/JumpGame.kt
package problems1_100

enum class JumpGame {
    DEFAULT {
        override fun canJump(nums: IntArray): Boolean {
            var goal = nums.size - 1
            for (i in nums.indices.reversed()) {
                if (i + nums[i] >= goal) {
                    goal = i
                }
            }
            return goal == 0
        }
    },
    ;

    abstract fun canJump(nums: IntArray): Boolean
}
